qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -196,6 +196,7 @@ def __init__(
         "This property will be deprecated in a future release and subsequently "
         "removed after that.",
         category=PendingDeprecationWarning,
```

Comment:
Not that it matters too much but all the algorithms were marked pending in 0.22. With the PR to change algorithms over from pending to fully deprecated, for the next release, I imagine this since would change as well (to 0.24.0) to reflect the change to the category. and be a base for that, rather than the version where things commenced with pending.

Reply:
Ah, as in, I can set these pending ones to be `since="0.24.0"`? I like that! 

It's better than my plan before to have the documentation plugin render the version `0.23.0_pending`. I think `0.24.0` is more clear than `0.23.0_pending`. And it avoids the deprecation author from needing to change `since=0.23.0` to `since=0.24.0` once they switch from pending to deprecated.

Reply:
I'm referring to #9532 of course. That will change things to deprecated such that its deprecated since 0.24.0. No docs should be published off any of this in the interim - so as long as the deprecated PR makes it in - have no reason to believe it won't though - things would be fine. It does make the change a kind of half-way house though - PendingDeprecation but with the version that the Deprecation is intended to commence at!

Reply:
That makes sense. This makes me realize these semantics are better for pending deprecations:

> If the deprecation is pending, set the version to when the deprecation will officially start.

That makes the `.. deprecated ::` directive have a more sensible version, e.g. we will show `0.24.0` rather than `0.23.0_pending`. And less work for authors when switching Pending because the `since` version won't need to change at all.

Reply:
I'm not sure that this is the best strategy - it could just be the responsibility of #9532 to update the version once it's certain.  The deprecations of these algorithms were already pushed back one minor version, and while there's no particular indication it'll change again, I think we probably ought to build the policy around "since" referring to the warning that's actually being emitted, not one that we expect to emit in the future.